Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Link/Link.cpp

```diff
@@ -282,6 +284,35 @@ namespace AzToolsFramework
             }
         }
 
+        PrefabOverridePrefixTree Link::GenerateOverrideSubTreeForEntity(
+            const PrefabDomValue& subpathPatches, const AZStd::string& pathToEntity)
+        {
+            PrefabOverridePrefixTree prefixTree;
+            if (subpathPatches.IsArray())
+            {
+                rapidjson::GenericArray patchesArray = subpathPatches.GetArray();
+                for (rapidjson::SizeType i = 0; i < patchesArray.Size(); i++)
+                {
+                    PrefabDom patchEntry;
+                    patchEntry.CopyFrom(patchesArray[i], patchEntry.GetAllocator());
+
+                    auto pathIter = patchEntry.FindMember("path");
+                    if (pathIter != patchEntry.MemberEnd())
+                    {
+                        AZStd::string subPath = pathIter->value.GetString(); // eg: '/Components/Component_[123]/Child Sort Order/0'
+                        AZStd::string fullPath = pathToEntity + subPath;
+                        pathIter->value.SetString(fullPath.c_str(), static_cast<rapidjson::SizeType>(fullPath.length()),
+                            patchEntry.GetAllocator());
+
+                        PrefabOverrideMetadata overrideMetadata(AZStd::move(patchEntry), m_patchIndexCounter++);
```

Comment:
By incrementing the patch counter here in the hopes of making an additional call later to use the same indices feels error prone to me. For example, what happens when you generate the subtree but then have some issues adding it back to the tree under the link object? In such cases, you would have incremented the counter unnecessarily. 

Instead of doing this way, maybe we can remove the overrides at the path first, then generate and insert at the same time to the tree?

Reply:
Yeah I feel it would be error prone as well. I used to add some logic to recalibrate index counter during tree rebuild but later changed to just increase the counter always.

Taking this as an example:
```
Index   0  1   2  3  4  5  6  |  7  8  9  10  11  12  13  14  15  |  16  17  18  19  20
Patch   [existing overrides]     [existing + new overrides]          [future overrides]
                                 (edit-entity node)                  (follow-up node)
```

(for the edit-entity node) When every of its undo/redo operations happens, the indices 7-15 won't conflict with other indices like 0-6 and 16-20. I am worried as you said if there is some issue generating the subtree that may break the consistency. But it looks like as we always increase the index counter, the future indices should always be greater than the one in the node and it won't cause any issue?

And note that the indices are automatically recalibrated (removing unused index gaps) every time we reload the level.

Reply:
Had offline discussion about this. There are two options here to avoid the disconnect between the time we update the index and when we actually insert patches into the tree.

option 1 : make m_patchIndexCounter an atomic variable so that this won't become an issue in the future due to asynchronous requests
option 2 : Instead of adding the patches in the Redo() function, do it in the Capture() function itself and skip the first redo. By doing so, you'll do both the patch index increment and patch addition in the same function.

Reply:
> option 2 : Instead of adding the patches in the Redo() function, do it in the Capture() function itself and skip the first redo. By doing so, you'll do both the patch index increment and patch addition in the same function.

I am experimenting with option 2 with a function called `AddOverrides(path, patches)` that is called in Capture. Looks like it is working as expected.